Cargo.toml

```diff
@@ -1,3 +1,22 @@
 [workspace]
 
 members = ["crates/*"]
+
+# Compile all dependencies with some optimizations when building this crate on debug
+# This slows down clean builds by about 50%, but the resulting binaries can be orders of magnitude faster
+# As clean builds won't occur very often, this won't slow down the development process
+[profile.dev.package."*"]
+opt-level = 2
+
+# Turn on a small amount of optimisation in development mode. This might interfere when trying to use a debugger
+# if the compiler decides to optimize some code away, if that's the case, it can be set to 0 or commented out
+[profile.dev]
+opt-level = 1
```

Comment:
what was the impact of this optimization? I'd prefer not to optimize our code, since that can be a difficult-to-understand problem as new contributors are onboarded

Reply:
Running the following operations with a vault of 7k items, using PBKDF2 with 600k rounds:
- Login with password:
  - All the optimizations: 1s
  - Only external dependencies: 6s
  - Previously: 12s

- Sync without decryption:
  - All the optimizations: 350ms
  - Only external dependencies: 670ms
  - Previously: 850ms

- Decryption of the items:
  - All the optimizations: 52ms
  - Only external dependencies: 276ms
  - Previously: 511ms

The biggest slow down when removing that optimization is during the login, entirely because of PBKDF2. This seems to be because those functions are defined as [inline] in the crate so the compiler doesn't run the optimizations that we have defined for external crates.

Reply:
I guess we could move the auth logic into a separate crate and optimize just that. I assume there's no way to tell rust to not inline the crate?

Reply:
I don't think we can move to a separate crate, we need kdf derivations in a few places (password reprompt, change password) so that can't be a long-term solution.

Let's just try and make this a known possibility like you've done here.